### PR TITLE
FEAT: added repo sync button and handlers

### PR DESCRIPTION
**Pull request by mailmesriza98**
closes https://github.com/Real-Dev-Squad/website-dashboard/issues/425
Description:
Implement a "Repo Sync" button on the website dashboard. This button will trigger the synchronization of the repository with the staging environment by calling the Sync API. This button is should be behind a feature flag

Screenshot 2023-07-29 135348

test

### CODE REVIEW

File: script.js [hunk 1]
```diff
@@ -117,6 +126,74 @@ async function handleSync(
...
+  toast.classList.remove('hidden');
+
+  if (type === 'success') {
+    for (let i = 0; i < message.merge_status.length; i++) {
+      if (message.merge_status[i].status.updated) {
+        let repo = message.merge_status[i].repository;
+        let text = repo.substring(repo.lastIndexOf('/') + 1) + ' synced';
+        toast.innerHTML = `<div class="message"> ✓ ${text}</div>`;
+      } else {
+        toast.innerHTML = `<div class="message">✓ All repos uptodate</div>`;
```
Comment (Pratiyushkumar) [inline script.js:142]: can we `All repos uptodate` as constant and use it here please

**Review by Pratiyushkumar — CHANGES_REQUESTED** [review]
Please follow proper format to raise PR by adding proper descriiption for reviewer and also write test and add test coverage to PR

File: style.css
Comment (sahsisunny) [inline style.css:1]: Could you please create the variables for the color and specify the usage of REM instead of PX in case the value is less than 10PX?

**Review by sahsisunny — DISMISSED** [review]

File: style.css [hunk 2]
```diff
@@ -239,3 +237,61 @@ footer {
...
+  top: 3rem;
+  right: 0.5rem;
+  padding: 10px;
+  border-radius: 0.3rem;
+  font-weight: bold;
+  border: 0.06rem solid;
+  display: flex;
+  align-items: center;
+  flex-direction: column;
+}
```
Comment (vinit717) [inline style.css:252]: Use px for smaller value

File: style.css [hunk 3]
```diff
@@ -239,3 +237,61 @@ footer {
...
+  color: var(--red-color);
+}
+
+.hidden {
+  display: none !important;
+}
+
+.progress-bar {
+  width: 100%;
+  height: 0.06rem;
```
Comment (vinit717) [inline style.css:271]: here also

File: style.css
Comment (vinit717) [inline style.css:1]: use px instead of rem for smaller value

Comment (sahsisunny) [reply style.css:1]: > use px instead of rem for smaller value

https://github.com/Real-Dev-Squad/website-dashboard/pull/427#discussion_r1278281034

**Review by sahsisunny — APPROVED** [review]

File: style.css [hunk 4]
```diff
@@ -239,3 +237,61 @@ footer {
...
+.success .progress-bar {
+  background-color: var(--success-color);
+}
+
+.failure .progress-bar {
+  background-color: var(--failure-color);
+}
+
+.fill {
+  height: 0.3rem;
```
Comment (vinit717) [inline style.css:285]: still, for small values, you are using rem

File: style.css [hunk 5]
```diff
@@ -239,3 +237,61 @@ footer {
     transform: translate(-50%, -50%) rotate(360deg);
   }
 }
+
+#toast {
+  position: absolute;
+  top: 3rem;
+  right: 0.5rem;
```
Comment (vinit717) [inline style.css:244]: +1

**Review by vinit717 — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]
lgtm

**Pull request merged**